RetailCoder.VBE/App.cs

```diff
@@ -56,36 +57,104 @@ public class App : IDisposable
             _configService = configService;
             _appMenus = appMenus;
             _stateBar = stateBar;
-            _hook = hook;
+            _indenter = indenter;
+            _hooks = hooks;
             _logger = LogManager.GetCurrentClassLogger();
 
-            _hook.KeyPressed += _hook_KeyPressed;
+            _hooks.MessageReceived += hooks_MessageReceived;
             _configService.SettingsChanged += _configService_SettingsChanged;
             _parser.State.StateChanged += Parser_StateChanged;
             _stateBar.Refresh += _stateBar_Refresh;
         }
 
-        private void _stateBar_Refresh(object sender, EventArgs e)
+        private Keys _firstStepHotKey;
+        private bool _isAwaitingTwoStepKey;
+        private bool _skipKeyUp;
+
+        private async void hooks_MessageReceived(object sender, HookEventArgs e)
+        {
+            if (sender is LowLevelKeyboardHook)
+            {
+                if (_skipKeyUp)
+                {
+                    _skipKeyUp = false;
+                    return;
+                }
+
+                if (_isAwaitingTwoStepKey)
+                {
+                    // todo: use _firstStepHotKey and e.Key to run 2-step hotkey action
+                    if (_firstStepHotKey == Keys.I && e.Key == Keys.M)
```

Comment:
I'd even go so far to try and get Keystrokes (or multi-step-keys) into a configuration based mapping, something like a properly encapsuled `Dictionary<Keystroke, Command>`. Hardcoding this doesn't seem like a viable option to me

Maybe that's only for a next step, though